src/utils/path.c

```diff
@@ -116,6 +116,27 @@ const char *path_c(const path *path) {
     return path->buf;
 }
 
+bool path_resolve(path *p) {
+#if defined(_WIN32) || defined(WIN32)
+    path tmp;
+    DWORD ret = GetFullPathNameA(p->buf, strlen(p->buf), tmp.buf, NULL);
+    if(ret > 0 && ret < PATH_MAX_LENGTH) {
+        *p = tmp;
+        normalize_slashes(p);
+        return true;
+    }
+    return false;
+#else
+    char *resolved = realpath(p->buf, NULL);
+    if(resolved == NULL) {
+        return false;
+    }
+    strncpy(p->buf, resolved, PATH_MAX_LENGTH);
```

Comment:
This is a misuse of strncpy, and does not guarantee null termination.

Use one of the openomf functions, like strncpy_or_abort.
Or, calculate the length of resolved, check that it fits in PATH_MAX_LENGTH (return false otherwise), and memcpy.

Reply:
We could probably just use strcpy here, since we know resolve is always null terminated.

Reply:
if(strlen(src) < MAX) strcpy() else abort()

Reply:
Oh, sure. I was just musing that there is actually a case where strcpy could be used safely.

Isn't strncpy_or_abort exactly the same though ? It does strlen and then does the copy; strlen and strcpy would do exactly the same.

Reply:
Anyways, swapped to strncpy_or_abort() now.

Reply:
> Isn't strncpy_or_abort exactly the same though ? It does strlen and then does the copy; strlen and strcpy would do exactly the same.

Yes and no: Yes, the result is the same. No, it is not exactly the same. strcpy has a memory-dependency where the moves depend on the values loaded, whereas in memcpy the memory-dependency is on the copy-size argument-- and if we need to debug a failure, having a central "strncpy_or_abort" implementation to add spew, breakpoints, see in stacktraces,.. is a boon.